### ISSUE TITLE
rpmdeps makes bad assumption when parsing ELF files

### ISSUE THREAD

**Issue by rossburton** [rpm-software-management/rpm#1887]
I experimented with the mold linker (https://github.com/rui314/mold) but we were getting packaging failures due to some libraries not having their Provides set correctly (ie libgmodule was no longer providing `libgmodule.so.1(64-bit)`).

It turns out that the cause was that `rpmdeps`  uses the presence of the `DT_DEBUG` dynamic entry to determine if a given object is a shared library or an executable, and mold was always writing an entry.

Since https://github.com/rui314/mold/commit/60c134893ca2349d33e9cac6b748425ebf5da3af mold only generates these for executables, but this is fundamentally a bad way to identify library vs executable.

The mold maintainer suggests instead:
> 1. Look for DT_FLAGS_1 dynamic section entry
> 2. Check if DF_1_PIE bit is set. If set, it's position-independent executable. Otherwise shared object file.
>
> If DT_FLAGS_1 entry is not present, I believe you can safely assume that that's a shared object file.

**Comment by pmatilai** [comment]
Yup, I know. The DT_DEBUG thing is an old hack (commit 83a0eb0b3633cdec627971d84dbbbb801870999c), better explained in elfutils tree:
https://sourceware.org/git/?p=elfutils.git;a=blob;f=src/elfclassify.c;hb=HEAD#l568

We should either call eu-elfclassify to determine the type or at least copy the much better heuristics from it.
